unblob/handlers/archive/arj.py

```diff
@@ -28,7 +32,12 @@ class ARJHandler(StructHandler):
         strings:
             $arj_magic = { 60 EA [5] 0? [2] 0? }
         condition:
-            $arj_magic
+            $arj_magic and
```

Comment:
I think we can do even more checks in Yara. If you take a look at http://www.opennet.ru/docs/formats/arj.txt, we can validate:

- host_os: `uint8(@arj_magic+offset) >= 0 and uint8(@arj_magic+offset) <= 9`
- file_type: `uint8(@arj_magic+offset) >= 0 and uint8(@arj_magic+offset) <= 4`

Reply:
I wanted to use the same logic as 7zip uses for verification. It does not limit neither HostOS nor FileType fileds, so I would keep it more permissive. Length & checksum checks should be enough to distinguish between garbage and normal ARJ chunk.